homeassistant/components/switchbot/coordinator.py

```diff
@@ -79,9 +77,9 @@ def _needs_poll(
 
     async def _async_update(
         self, service_info: bluetooth.BluetoothServiceInfoBleak
-    ) -> dict[str, Any]:
+    ) -> None:
         """Poll the device."""
-        return await self.device.update()
+        await self.device.update()
```

Comment:
`self.device.update()` doesn't return anything and the `data` attribute isn't used. Instead the `device` attribute is accessed directly. Furthermore, the `update()` method is typed with `-> None` upstream.
https://github.com/Danielhiversen/pySwitchbot/blob/0.34.1/switchbot/devices/device.py#L528